### ISSUE TITLE
Naming standards

### ISSUE THREAD

**Issue by bjornallvin** [DPC-Nordics/dpc-next#27]
## Pull requests:

**Comment by bjornallvin** [comment]
This is a suggested naming convention. I'm not married to any of it. Please comment or suggest changes, additions or comments or  questions.

**Comment by Juliafalk** [comment]
Looks good! However, I am wondering what to use when fixing multiple types in one branch? E.g. if doing styling but also refactoring some code at the same time?

**Comment by GuptaSiddhant** [comment]
I like the possible types but I would recommend against putting issue number before the slash.

I suggest following format for branch names:

```
<type>/<issue-number>-<short-description>
```

This has benefits:
- Categorisation of all branches of similar type.
- Follows standard practice (git-flow)
- Matches to [Conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)

**Comment by GuptaSiddhant** [comment]
> Looks good! However, I am wondering what to use when fixing multiple types in one branch? E.g. if doing styling but also refactoring some code at the same time?

Normally if the change is big, the convention is to go with `feat` or `fix` depending on what suits better.

**Comment by GuptaSiddhant** [comment]
For PR message, we should follow GitHub standards for linking issues:
https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

They require `<KEYWORD> #<ISSUE NUMBER>` somewhere in pr-message, description or commit-messages.

**Comment by bjornallvin** [comment]
> > Looks good! However, I am wondering what to use when fixing multiple types in one branch? E.g. if doing styling but also refactoring some code at the same time?
> 
> Normally if the change is big, the convention is to go with `feat` or `fix` depending on what suits better.

Yes, feat or fix trumphs the other types. They basically include all other types of changes.

**Comment by bjornallvin** [comment]
Also, it is not super important to follow the naming standard on commits to the feature branches. As when we merge the PR, we squash all commits into a new single commit onto main branch. It is that merge commit that is important to get according to standard.

**Comment by bjornallvin** [comment]
> For PR message, we should follow GitHub standards for linking issues: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
> 
> They require `<KEYWORD> #<ISSUE NUMBER>` somewhere in pr-message, description or commit-messages.

Either add a message in the merge commit descrtiption or link them manually. I guess both work.
